Update typed property array => ArrayCollection to prevent errors

johanadivare opened the pull request:
When using strict type checking you could get errors like:
```
TypeError: Cannot assign Doctrine\Common\Collections\ArrayCollection to property App\Entity\Category::$products of type array
```

xabbuh commented on doctrine/associations.rst:
```diff
@@ -220,7 +220,7 @@ class that will hold these objects:
 
             #[ORM\OneToMany(targetEntity: Product::class, mappedBy: 'category')]
-            private array $products;
+            private ArrayCollection $products;
```
```suggestion
            private Collection $products;
```

xabbuh replied:
because when loaded from the database the property will be populated with a `PersistentCollection` by Doctrine

johanadivare replied:
Ah yes and that implements `Collection` interface agreed.

JohJohan committed:
Update typed property array => ArrayCollection to prevent errors

When using strict type checking you could get errors like:
```
TypeError: Cannot assign Doctrine\Common\Collections\ArrayCollection to property App\Entity\Category::$products of type array
```

Co-authored-by: Christian Flothmann <christian.flothmann@gmail.com>

javiereguiluz commented:
Good catch, thanks @johanadivare.
